mp2-v1/src/values_extraction/merge.rs

```diff
@@ -0,0 +1,212 @@
+        _builder_parameters: Self::CircuitBuilderParams,
+    ) -> Self {
+        let table_a = PublicInputs::new(&verified_proofs[0].public_inputs);
+        let table_b = PublicInputs::new(&verified_proofs[0].public_inputs);
+        MergeTable::build(builder, table_a, table_b)
+    }
+
+    fn assign_input(
+        &self,
+        inputs: Self::Inputs,
+        pw: &mut PartialWitness<GFp>,
+    ) -> anyhow::Result<()> {
+        inputs.assign(pw, &self);
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::iter::once;
+
+    use super::*;
+    use mp2_common::{
+        group_hashing::{field_hashed_scalar_mul, weierstrass_to_point as wp},
+        keccak::PACKED_HASH_LEN,
+        rlp::MAX_KEY_NIBBLE_LEN,
+        C, D, F,
+    };
+    use mp2_test::circuit::{run_circuit, UserCircuit};
+    use plonky2::{
+        field::types::Sample,
+        iop::witness::{PartialWitness, WitnessWrite},
+    };
+    use plonky2_ecgfp5::curve::curve::Point;
+
+    use super::MergeTableWires;
+
+    #[derive(Clone, Debug)]
+    struct TestMerge {
+        merge: MergeTable,
+        table_a: Vec<F>,
+        table_b: Vec<F>,
+    }
+
+    impl UserCircuit<F, D> for TestMerge {
+        type Wires = (MergeTableWires, Vec<Target>, Vec<Target>);
+
+        fn build(b: &mut CBuilder) -> Self::Wires {
+            let table_a = b.add_virtual_targets(PublicInputs::<Target>::TOTAL_LEN);
+            let table_b = b.add_virtual_targets(PublicInputs::<Target>::TOTAL_LEN);
+            let wires =
+                MergeTable::build(b, PublicInputs::new(&table_a), PublicInputs::new(&table_b));
+            (wires, table_a, table_b)
+        }
+
+        fn prove(&self, pw: &mut PartialWitness<F>, wires: &Self::Wires) {
+            self.merge.assign(pw, &wires.0);
+            pw.set_target_arr(&wires.1, &self.table_a);
+            pw.set_target_arr(&wires.2, &self.table_b);
+        }
+    }
+
+    fn random_field_vector(n: usize) -> Vec<F> {
+        (0..n).map(|_| F::rand()).collect()
+    }
+
+    fn random_public_input(root_hash: Option<Vec<F>>) -> Vec<F> {
+        let h = root_hash.unwrap_or_else(|| random_field_vector(PACKED_HASH_LEN));
+        let k = random_field_vector(MAX_KEY_NIBBLE_LEN);
+        let t = F::NEG_ONE;
+        let dv = Point::rand().to_fields();
+        let dm = Point::rand().to_fields();
+        let n = F::from_canonical_u8(10);
+        h.into_iter()
+            .chain(k)
+            .chain(once(t))
+            .chain(dv)
+            .chain(dm)
+            .chain(once(n))
+            .collect()
+    }
+
+    #[test]
+    fn test_merge_table() {
+        let table_a = random_public_input(None);
+        let table_a_pi = PublicInputs::new(&table_a);
+        // making sure they both share the same root hash
+        let table_b = random_public_input(Some(table_a_pi.root_hash_info().to_vec()));
+        let table_b_pi = PublicInputs::new(&table_b);
+        let is_table_a_multiplier = true;
```

Comment:
What about making it an input parameter of this method and test both `true` and `false` values for this flag?